### PR TITLE
Grid Component

### PR DESCRIPTION
**Pull request by clloyd**
screen shot 2017-02-03 at 15 25 00

This adds a grid component complete with tests. 

Also adds a `./scripts/start.sh` startup for both play and fe asset compilation

### CODE REVIEW

File: public/js/components/FormFields/FormFieldImageSelect.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    window.removeEventListener('message', this.onMessage, false);
+  }
+
+  openModal = () => {
+      this.setState({ modalOpen: true });
+      window.addEventListener('message', this.onMessage, false);
+  }
+
+  validMessage(data) {
+    return data && data.crop && data.crop.data && data.image && data.image.data;
```
Comment (ShaunYearStrong) [inline public/js/components/FormFields/FormFieldImageSelect.js]: I hate JS sometimes

File: public/js/components/FormFields/FormFieldImageSelect.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+          </button>
+        </div>
+      </div>
+    )
+  }
+
+  render () {
+    return (
+      <div>
+        <label className="form__label" htmlFor={this.props.fieldName}>{this.props.fieldLabel}</label>
```
Comment (ShaunYearStrong) [inline public/js/components/FormFields/FormFieldImageSelect.js]: I've just realised I didn't do this on mine, but the value of `htmlFor` should be the same as the `id` of the element we're referring to.

File: public/js/util/imageHelpers.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+export function parseImageFromGridCrop(cropData) {
+  return {
+    assets: cropData.assets.map(parseAsset),
+    master: parseAsset(cropData.master),
+    mediaId: cropData.specification.uri
+  }
+}
+
+export function findSmallestAsset(assetsArray) {
+  return assetsArray.reduce((smallestAsset, newAsset) => {
```
Comment (ShaunYearStrong) [inline public/js/util/imageHelpers.js]: Couldn't we use `Array.prototype.sort()` instead? https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

Comment (ShaunYearStrong) [inline public/js/util/imageHelpers.js]: Ignore that, looks like this way is faster https://jsperf.com/reduce-vs-sort

Comment (clloyd) [inline public/js/util/imageHelpers.js]: Yeah reduce has a complexity of `n` vs sort which can be a lot more

File: public/js/components/FormFields/FormFieldImageSelect.js (hunk 2 above)
Comment (clloyd) [inline public/js/components/FormFields/FormFieldImageSelect.js]: Fixing

**Pull request merged**